### PR TITLE
Added console_scripts to the package.

### PR DESCRIPTION
**Pull request by tokestermw**
Added the Python scripts in `bin/` to `setup.py` so that they are callable

Decided to prepend the scripts with `opennmt` instead of `bin` since it's more descriptive.

Install package first by `python setup.py develop`.

Example:

```
opennmt.main  # this is equivalent to python -m bin.main
```

Related issue: https://github.com/OpenNMT/OpenNMT-tf/issues/93

Edit: Will probably need some edits to the `README` as well.

### CODE REVIEW

**Comment by guillaumekln** [comment]
Thank you for looking into this.

Does that work even if `bin/` is excluded from the package? Don't we need to move `bin/` inside the `opennmt/` directory?

**Comment by tokestermw** [comment]
Ah, you're correct. Pip installing directly fails to load the `bin` directory.

```
pip install git+https://github.com/tokestermw/OpenNMT-tf@add-command-line-to-package
```

So the least disruptive way is to move the `bin` dir inside `opennmt`, remove the `exclude`, and add to `console_scripts`:

```
        "console_scripts": [
            "bin.ark_to_records=opennmt.bin.ark_to_records:main",
        ...
```

Does this sound good?

File: setup.py [hunk 1]
```diff
@@ -15,5 +15,16 @@
         "nose2"
     ],
     test_suite="nose2.collector.collector",
-    packages=find_packages(exclude=["bin", "*.tests"])
+    packages=find_packages(exclude=["*.tests"]),
+    entry_points={
+        "console_scripts": [
+            "bin.ark_to_records=opennmt.bin.ark_to_records:main",
```
Comment (guillaumekln) [inline setup.py]: Let's improve all script names:

* Remove `bin.`
* Prefix with `onmt-`
* Replace `_` by `-`

e.g. `onmt-ark-to-records`.

File: setup.py [hunk 2]
```diff
@@ -15,5 +15,16 @@
         "nose2"
     ],
     test_suite="nose2.collector.collector",
-    packages=find_packages(exclude=["bin", "*.tests"])
+    packages=find_packages(exclude=["*.tests"]),
```
Comment (guillaumekln) [inline setup.py]: As per the comment above, `bin` should be added back.

**Comment by tokestermw** [comment]
Sure, I can make those changes.

It may make sense to deprecate the top-level `bin/` eventually since it adds a bit of code redundancy (you can still call directly by `python opennmt/bin/build_vocab.py`).

Edit: changes are made! You can test by:

```
python bin/ark_to_records.py
onmt-ark-to-records

python bin/average_checkpoints.py
onmt-average-checkpoints

python bin/build_vocab.py
onmt-build-vocab

python bin/detokenize_text.py yoyo
onmt-detokenize-text yoyo

python bin/main.py
onmt-main

python bin/merge_config.py
onmt-merge-config

python bin/tokenize_text.py yoyo
onmt-tokenize-text yoyo
```

**Comment by guillaumekln** [comment]
Thanks! That is really helpful.

**Pull request merged**